src/app/redux/history/reducer.js

```diff
@@ -0,0 +1,51 @@
+export const initialState = {
+  history: [],
+};
+
+export function historyReducer(state = initialState, action) {
+  switch (action.type) {
+    case 'GET_HISTORIES':
+      return state.history;
+    case 'SET_HISTORY':
+      return {
+        ...state,
+        history: [...state.history, action.payload],
+      };
+    case 'EDIT_EXPRESSION_HISTORY': {
+      let {newExpression, id} = action.payload;
+      let historyCopy = [];
+      historyCopy = [...state.history];
+      let historyFiltered = [];
+      historyCopy.map((expression, key) => {
+        if (key !== id) {
+          historyFiltered.push(expression);
+        } else {
+          historyFiltered.push(newExpression);
+        }
+      });
+      return {
+        ...state,
+        history: historyFiltered,
+      };
+    }
+    case 'DELETE_HISTORY_FOR_ID': {
+      let historyCopy = [];
+      historyCopy = [...state.history];
```

Comment:
Same here. Also, there is no need here since filter doesn't modify the original array, it creates a new one.

Reply:
Done